### PR TITLE
build(k8s): To fix issue for make generate-k8s-api

### LINKED ISSUE #10946: Add checker for out-of-sync generated.deepcopy.go files
This simple script should be enough to detect if zz_generated.deepcopy files are out of sync.

`find . -not -path "./vendor/*" -iname "zz_generated.deepcopy.go" -exec rm {} \; && make generate-k8s-api && git diff`

### PR DESCRIPTION
**Pull request by sayboras**
There is below error while running make generate-k8s-api in vagrant box
    
<details>
<summary>Make generate-k8s-api in vagrant box</summary>

```
    vagrant@runtime1:~/go/src/github.com/cilium/cilium$ make generate-k8s-api
    2020/05/11 13:13:25 Unable to add directory
    "k8s.io/apimachinery/pkg/apis/testapigroup/v1": unable to import
    "k8s.io/apimachinery/pkg/apis/testapigroup/v1": cannot find package
    "k8s.io/apimachinery/pkg/apis/testapigroup/v1" in any
    of:
 /home/vagrant/go/src/github.com/cilium/cilium/vendor/k8s.io/apimachinery/pkg/apis/testapigroup/v1
    (vendor
    tree)
            /usr/local/go/src/k8s.io/apimachinery/pkg/apis/testapigroup/v1
    (from
    $GOROOT)
            /home/vagrant/go/src/k8s.io/apimachinery/pkg/apis/testapigroup/v1
    (from $GOPATH)
    exit status 1
    Makefile:368: recipe for target 'generate-k8s-api' failed
    make: *** [generate-k8s-api] Error 1
```

</details>

### Approach:
    
-   Remove un-used package k8s.io/apimachinery/pkg/apis/testapigroup/v1
-   Add GOPATH in Vagrantfile
-   Rename go-mod github action to go-check, and add the job to check if k8s generated files are in sync

    Closes #10946
    Signed-off-by: Tam Mach <sayboras@yahoo.com>

```release-note
Fix make generate-k8s-api
```

### CODE REVIEW

File: .github/workflows/k8s-generated.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+          path: ./src/github.com/${{ github.repository }}
+      - name: Install protobuf dependencies
+        run: |
+          sudo apt install -y protobuf-compiler libprotobuf-dev
+      - name: Check generated files
+        env:
+          GOPATH: ${{ runner.workspace }}
+        run: |
+          make generate-k8s-api
+          git diff --exit-code
```
Comment (pchaigno) [inline .github/workflows/k8s-generated.yaml]: We're having more and more GitHub Actions, we should start thinking about grouping them. Maybe we could add this job to the go-mod workflow? Does that make sense?

File: .github/workflows/k8s-generated.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+jobs:
+  generate-k8s-api:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.14.2
+      - name: Checkout code
+        uses: actions/checkout@v1
```
Comment (pchaigno) [inline .github/workflows/k8s-generated.yaml]: We've tried to use `v2` in other GitHub Actions.

File: .github/workflows/k8s-generated.yaml (hunk 1 above)
Comment (sayboras) [reply .github/workflows/k8s-generated.yaml]: yup totally making sense to me :), so let me group all go related action together.

File: .github/workflows/k8s-generated.yaml (hunk 2 above)
Comment (sayboras) [reply .github/workflows/k8s-generated.yaml]: oh let me bump this as well, this was copied from my old configuration.

File: .github/workflows/go-check.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.14.2
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          path: src/github.com/${{ github.repository }}
+      - name: Install protobuf dependencies
+        run: |
+          sudo apt install -y protobuf-compiler libprotobuf-dev
```
Comment (aanm) [inline .github/workflows/go-check.yaml]: Might be worth set a specific protobuf version.

File: .github/workflows/go-check.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      - name: Set GOPATH
+        run: |
+          echo "##[set-env name=GOPATH;]$GITHUB_WORKSPACE"
+        shell: bash
+      - name: Check k8s generated files
+        run: |
+          cd src/github.com/${{ github.repository }}
+          make generate-k8s-api
```
Comment (aanm) [inline .github/workflows/go-check.yaml]: This step is not enough, one needs to delete all `zz_generated.deepcopy.go` files

**Review by aanm — CHANGES_REQUESTED** [review]

File: .github/workflows/go-check.yaml (hunk 4 above)
Comment (sayboras) [reply .github/workflows/go-check.yaml]: I assume deletion will help to detect dangling files i.e. the file should be not generated but already checked in ?

File: .github/workflows/go-check.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
+name: Go related checks
```
Comment (pchaigno) [inline .github/workflows/go-check.yaml]: Nit: This should be `Go-related checks`. No need to send an update just for that though :-)

**Review by pchaigno — APPROVED** [review]
LGTM assuming @aanm's comments are addressed.

File: .github/workflows/go-check.yaml (hunk 4 above)
Comment (aanm) [reply .github/workflows/go-check.yaml]: @sayboras I just remember, there's a script doing this :facepalm: https://github.com/cilium/cilium/blob/master/contrib/scripts/check-k8s-code-gen.sh feel free to use the script or to delete if you embed the code into the github action.

>I assume deletion will help to detect dangling files i.e. the file should be not generated but already checked in ?

yes that's correct

Comment (sayboras) [reply .github/workflows/go-check.yaml]: awesome, I will re-use the script, as other might run this script locally before pushing

File: .github/workflows/go-check.yaml (hunk 3 above)
Comment (sayboras) [reply .github/workflows/go-check.yaml]: Sure, I updated with the same version in vagrant box, hope it is fine :)

Comment (sayboras) [reply .github/workflows/go-check.yaml]: @aanm as discussed, I have updated protobuf version to 3.11.4. Please let me know if you have other comment.

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
restest-runtime

**Comment by aanm** [comment]
restest-4.19

**Pull request merged**